src/language-server/minilogo-validator.ts

```diff
@@ -1,11 +1,11 @@
-import { ValidationAcceptor, ValidationCheck, ValidationRegistry } from 'langium';
-import { Def, MiniLogoAstType, Model } from './generated/ast';
+import { AstNode, ValidationAcceptor, ValidationCheck, ValidationRegistry } from 'langium';
+import { isDef, isModel, MiniLogoAstType } from './generated/ast';
 import type { MiniLogoServices } from './minilogo-module';
 
 /**
  * Map AST node types to validation checks.
  */
-type MiniLogoChecks = { [type in MiniLogoAstType]?: ValidationCheck | ValidationCheck[] }
+type MiniLogoChecks = { [type in (keyof MiniLogoAstType)]?: ValidationCheck | ValidationCheck[] }
```

Comment:
Suggestion: There is now a builtin type for this
```suggestion
type MiniLogoChecks = ValidationChecks<MiniLogoAstType>
```

Reply:
Fantastic! Let's take it.